src/utils/params/requestParams.ts

```diff
@@ -1,7 +1,19 @@
 const DEFAULT_PAGE_SIZE = 100;
 
+export type Criteria<R> = Record<string, string> & { type?: R };
```

Comment:
Maybe let's add an underscore or two to the `type?` (`__type?`) and leave a comment that explicitly says it's not meant to be used and it's only there because typescript doesn't enforce generic type otherwise.